pkgs/applications/virtualization/libgovirt/default.nix

```diff
@@ -1,46 +1,46 @@
 { lib
 , stdenv
 , fetchurl
-, glib
 , gnome
-, librest
-, libsoup
+, meson
 , pkg-config
 , gobject-introspection
+, ninja
+, librest_1_0
 }:
 
 stdenv.mkDerivation rec {
   pname = "libgovirt";
-  version = "0.3.8";
+  version = "0.3.9";
 
   outputs = [ "out" "dev" ];
 
   src = fetchurl {
     url = "mirror://gnome/sources/libgovirt/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
-    sha256 = "HckYYikXa9+p8l/Y+oLAoFi2pgwcyAfHUH7IqTwPHfg=";
+    sha256 = "sha256-sS/DV58tAH5D/vHcwPSz+MEHDC+2o/WLI8U8dmJDIXw=";
   };
 
-  patches = lib.optionals stdenv.isDarwin [
-    # The flag breaks the build on darwin and doesn't seem necessary
-    ./no-version-script-ld-flag.patch
+  # https://gitlab.gnome.org/GNOME/libgovirt/-/issues/9
+  patches = [
+    ./auto-disable-incompatible-compiler-warnings.patch
   ];
+  NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isClang (lib.concatStringsSep " " [
+    "-Wno-typedef-redefinition"
+    "-Wno-missing-field-initializers"
+    "-Wno-cast-align"
+  ]);
 
   nativeBuildInputs = [
+    meson
     pkg-config
     gobject-introspection
-  ];
-
-  buildInputs = [
-    libsoup
+    ninja
   ];
 
   propagatedBuildInputs = [
-    glib
```

Comment:
Why is this no longer propagated?

It is better not to rely on propagated dependencies of dependencies for explicit dependencies, since the former can change.

Reply:
The dep doesn't seem necessary anymore; it builds without.

Reply:
That is because librest also propagates it. But the pkg-config files still lists is in `Requires` so it should be propagated here.

Reply:
Ah, so glib is propagated from librest, so it works but only as long as librest propagates the dep?

Reply:
Which pkg-config are you referencing?

Reply:
`$dev/lib/pkgconfig/govirt-1.0.pc`.